core/types/permissions_cache.go

```diff
@@ -110,77 +110,126 @@ type PermissionConfig struct {
 	SubOrgBreadth *big.Int         `json:"subOrgBreadth"`
 }
 
+var syncStarted = false
+
+var DefaultAccess = FullAccess
+var QIP714BlockReached = false
+var networkAdminRole string
+var orgAdminRole string
+
+const defaultOrgMapLimit = 2000
+const defaultRoleMapLimit = 2500
+const defaultNodeMapLimit = 1000
+const defaultAccountMapLimit = 6000
+
+var OrgInfoMap = NewOrgCache()
+var NodeInfoMap = NewNodeCache()
+var RoleInfoMap = NewRoleCache()
+var AcctInfoMap = NewAcctCache()
+
+
 type OrgKey struct {
 	OrgId string
 }
 
-type NodeKey struct {
-	OrgId string
-	Url   string
+type OrgCache struct {
+	c                 *lru.Cache
+	mux               sync.Mutex
+	evicted           bool
+	populateCacheFunc func(orgId string) *OrgInfo
+}
+
+
+func (o *OrgCache) PopulateCacheFunc(cf func(string) *OrgInfo) {
+	o.populateCacheFunc = cf
+}
+
+func NewOrgCache() *OrgCache {
+	orgCache := OrgCache{evicted: false}
+	onEvictedFunc := func(k interface{}, v interface{}) {
+		orgCache.evicted = true
+	}
+	orgCache.c, _ = lru.NewWithEvict(defaultOrgMapLimit, onEvictedFunc)
+	return &orgCache
 }
 
 type RoleKey struct {
 	OrgId  string
 	RoleId string
 }
 
-type AccountKey struct {
-	AcctId common.Address
+type RoleCache struct {
+	c                 *lru.Cache
+	evicted           bool
+	populateCacheFunc func(*RoleKey) *RoleInfo
 }
 
-type OrgCache struct {
-	c   *lru.Cache
-	mux sync.Mutex
+func (r *RoleCache) PopulateCacheFunc(cf func(*RoleKey) *RoleInfo) {
+	r.populateCacheFunc = cf
 }
 
-type NodeCache struct {
-	c *lru.Cache
+func NewRoleCache() *RoleCache {
+	roleCache := RoleCache{evicted: false}
+	onEvictedFunc := func(k interface{}, v interface{}) {
+		roleCache.evicted = true
+	}
+	roleCache.c, _ = lru.NewWithEvict(defaultRoleMapLimit, onEvictedFunc)
```

Comment:
handle error returned by NewWithEvict

Reply:
same as earlier response. NewWithEvict returns error only when the size passed is zero which is not the case.